### PR TITLE
Use PL logger instance

### PR DESCRIPTION
**Pull request by AlbertMitjans**
Stops using the root logger instance. See:  
- #2006 
- #3731 

Fixes: https://github.com/PennyLaneAI/pennylane/issues/3731

### CODE REVIEW

**Comment by timmysilv** [comment]
Love the fix chosen 😄 could you remove the 2 mentions of logging in the docstrings of tracker.py? a user might read it and think we support logging, but we do not. with that, all LGTM!

**Comment by AlbertMitjans** [comment]
> a user might read it and think we support logging, but we do not.

The purpose of this PR is to allow users to use `logging`, right? Thus there should be no harm if they use `logging` in the callable of the `Tracker` class. Or am I missing sth? 🤔

**Review by timmysilv — APPROVED** [review]
ahh, you mean logging set up by the user, not done by pennylane itself, and the tracker means this too. good for me

File: pennylane/configuration.py
Comment (josh146) [inline pennylane/configuration.py:56]: Will this be visible on import of PL? I have a feeling that's why we had `logger.info` instead of `logger.warn` previously. @trbromley

Comment (trbromley) [reply pennylane/configuration.py:56]: +1 not to have this show up on import. Can we limit the scope of the warning?

Comment (AlbertMitjans) [reply pennylane/configuration.py:56]: Sorry for not mentioning this change. I saw your thumbs up in [this comment](https://github.com/PennyLaneAI/pennylane/issues/2006#issuecomment-1009413835) and assumed we could delete it.

This definitely raises an error when importing PL... I'll keep using `logging` then, but use our own logger instance.

Comment (AlbertMitjans) [reply pennylane/configuration.py:56]: @trbromley maybe using a [ResourceWarning](https://docs.python.org/3/library/exceptions.html#ResourceWarning)?

This warning is only shown in Python Development Mode. My question is: will users ever use this mode? If they will (most probably) never see the warning, should we just remove it?

Comment (josh146) [reply pennylane/configuration.py:56]: Yeah, good point @AlbertMitjans. Another option is to use `log = logging.getLogger(__name__)` here, so that this module is not using or getting the root logger 🤔 Although I don't know if that has implications for the users reporting those bugs.

Comment (AlbertMitjans) [reply pennylane/configuration.py]: I used [this user's suggestion](https://github.com/PennyLaneAI/pennylane/issues/3731#issue-1574552685): `log = logging.getLogger(__name__)`.

I'm not super familiar with `logging`, but will users ever see this `info` message? Don't we need a handler to display the log messages?

**Comment by timmysilv** [comment]
I'm getting a bit of whiplash now. What exactly is the value of that warning if no config file is found? who should see it and when? I was kinda excited to get rid of all uses of `logging` 🙁

File: pennylane/configuration.py
Comment (AlbertMitjans) [reply pennylane/configuration.py]: FYI, I believe the `info` message was never shown to users, because the root logger is instantiated with level WARNING, thus all INFO messages are ignored:
```pycon
>>> import logging
>>> logging.info("hello world")  # nothing is shown/logged
```
Consequently, I believe we can definitely remove this info message without changing any code behaviour.

File: pennylane/configuration.py
Comment (AlbertMitjans) [inline pennylane/configuration.py:25]: This had no effect 😵‍💫

**Comment by AlbertMitjans** [comment]
I decided to remove the message because I believe it was never logged ([see message](https://github.com/PennyLaneAI/pennylane/pull/3738#discussion_r1100416586)). I will wait for another approval before merging.

**Comment by leofang** [comment]
Thanks all!

**Pull request merged**
